internal/cmd/run.go

```diff
@@ -79,6 +93,20 @@ func runCmd() *cobra.Command {
 				}
 			}
 
+			for _, block := range runBlocks {
+				if block.PromptEnv() {
+					varPrompts := getCommandExportExtractMatches(block.Lines())
+					for _, ev := range varPrompts {
+						newVal, err := promptForEnvVar(cmd, ev)
+						block.SetLine(ev.LineNumber, replaceVarValue(ev, newVal))
```

Comment:
Modifying `lines` may cause issues, because for example in TUI, the same script may be run multiple times. This will persist the environment variables instead of letting the user select each time.

I think we should implement a `Clone` method on `document.CodeBlock` so that you don't modify the underlying code block.